socorro/cron/jobs/update_signatures.py

```diff
@@ -48,9 +53,41 @@ class UpdateSignaturesCronApp(BaseCronApp):
         doc='Print to stdout instead of updating/inserting data'
     )
 
+    def update_crashstats_signature(self, connection, signature, report_date, report_build):
+        # Pull the data in the db. If it's there, then do an update. If it's
+        # not there, then do an insert.
+        try:
+            sig = single_row_sql(
+                connection,
+                """
+                SELECT signature, first_build, first_date
+                FROM crashstats_signature
+                WHERE signature=%s
+                """,
+                (signature,)
+            )
+            sql = """
+            UPDATE crashstats_signature
+            SET first_build=%s, first_date=%s
+            WHERE signature=%s
+            """
+            params = (
+                min(sig[1], report_build),
+                min(str(sig[2]), str(report_date)),
+                sig[0]
+            )
+        except SQLDidNotReturnSingleRow:
+            sql = """
+            INSERT INTO crashstats_signature (signature, first_build, first_date)
+            VALUES (%s, %s, %s)
+            """
+            params = (signature, report_build, report_date)
+
+        execute_no_results(connection, sql, params)
```

Comment:
The original does an "upsert", but I chose to do a basic implementation instead of an upsert because this is going to get converted to Django queryset stuff once we get crontabber into Django-land.